### PR TITLE
Feature changes

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
 - Changed the displayed restaurant map from storing names to object Ids, which crashed the code, so I changed it to Yelp ID which works well
 - Changed the source of the list for map activity from the explore fragment to itself, which slowed stuff down, but is more reliable
 - Added all the loading spinners to the app, and makes app look a lot less sporadic.

### CODE REVIEW

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java [hunk 1]
```diff
@@ -69,11 +71,14 @@ public void dataChanged() {
...
                     public void run() {
                         adapter.notifyDataSetChanged();
+                        rvRestaurants.setVisibility(View.VISIBLE);
+                        explore_progress_bar.setVisibility(View.GONE);
                     }
                 });
             }
         };
         swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+        explore_progress_bar = view.findViewById(R.id.explore_progress_bar);
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ExploreFragment.java:81]: camel case

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/FindFriendsFragment.java [hunk 2]
```diff
@@ -31,6 +32,7 @@ public class FindFriendsFragment extends Fragment {
     private RecyclerView rvUsers;
     private FindFriendsAdapter adapter;
     private List<FriendObservable> allUsers;
+    private ProgressBar search_progress_bar;
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/FindFriendsFragment.java:35]: camel case

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/FindFriendsFragment.java [hunk 3]
```diff
@@ -55,12 +57,15 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         super.onViewCreated(view, savedInstanceState);
         searchFriends = view.findViewById(R.id.searchFriends);
         rvUsers = view.findViewById(R.id.rvUsers);
+        search_progress_bar = view.findViewById(R.id.pbSearchFriends);
+        search_progress_bar.setVisibility(View.GONE);
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/FindFriendsFragment.java:61]: camel case

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java [hunk 4]
```diff
@@ -74,6 +75,7 @@ public class ProfileFragment extends Fragment implements Observer, View.OnClickL
     private ImageView ivFindFriends;
     private ImageView ivFollow;
     private ImageView ivLock;
+    private ProgressBar profile_progress_bar;
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:78]: camel case

**Review by asaxena1312 — CHANGES_REQUESTED** [review]
Please work on camel casing and send back for review :)

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 5]
```diff
@@ -28,24 +29,29 @@
...
 import com.parse.ParseUser;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class MapActivity extends AppCompatActivity implements OnMapReadyCallback{
     private List<RestaurantObservable> restaurantList;
     private UserObservable user;
     private String TAG = "Map Activity";
+    private int max_radius = 40000;
```
[the remark this answers is not in the document]
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java:44]: done

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 6]
```diff
@@ -28,24 +29,29 @@
...
 
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_map);
-        restaurantList = getIntent().getParcelableArrayListExtra("restaurants");
-        Log.i(TAG,String.valueOf(restaurantList.size()));
+        //restaurantList = getIntent().getParcelableArrayListExtra("restaurants");
```
[the remark this answers is not in the document]
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java:51]: done

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 7]
```diff
@@ -90,4 +96,25 @@ public boolean onMarkerClick(Marker marker) {
             }
         });
     }
+
+    private void parseGeneral() {
+        ParseQuery<Restaurant> query = ParseQuery.getQuery(Restaurant.class);
+        if (user.getCoordinates() != null) {
+            query.whereWithinKilometers("location_coordinates", user.getCoordinates(), max_radius/1000);
```
[the remark this answers is not in the document]
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java:103]: done

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 8]
```diff
@@ -90,4 +96,25 @@ public boolean onMarkerClick(Marker marker) {
...
+        } else {
+            query.whereEqualTo("city", user.getCity());
+            query.whereEqualTo("state", user.getState());
+        }
+        query.setLimit(20);
+        List<RestaurantObservable> observed = new ArrayList<RestaurantObservable>();
+        try {
+            List<Restaurant> parse_restaurants = query.find();
```
[the remark this answers is not in the document]
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java:111]: done

**Review by nammyfb — APPROVED** [review]

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 9]
```diff
@@ -64,7 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onMapReady(GoogleMap googleMap) {
         // Add a marker in Sydney, Australia,
```
[the remark this answers is not in the document]
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java]: Yeah deleted that!

**Pull request merged**